### ISSUE TITLE
[FeatureRequest] make `require("yargs").default` as same as `require("yargs")`

### ISSUE THREAD

**Issue by loynoir** [yargs/yargs#1949]
### Brief
`strictOptions` can not be trigger under `require("yargs").default`

### Background
[official document](https://github.com/yargs/yargs#esm)

```ts
// x.ts
import yargs from 'yargs'
```

```js
// tsc output x.js is something like
const yargs = require("yargs").default
```

### Expected
```js
require("yargs").default(["bar", "-a", "10"]).strictOptions().argv
```
```txt
// Error
Options:
  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]

Unknown argument: a
```

### Actual
```js
require("yargs").default(["bar", "-a", "10"]).strictOptions().argv
```
```txt
// Error is not trigger
{
  '10': undefined,
  _: [],
  bar: undefined,
  '-a': undefined,
  a: undefined,
  '$0': ''
}
```

### Workaround
```ts
// y.ts
import * as yargs from 'yargs'
```

```js
// tsc output y.js is something like
var yargs = require("yargs");
```

### Reproduce Gist
[gist](https://gist.github.com/loynoir/580bb181222bf21452412c0a86ae9cbf)

**Comment by karizma** [comment]
The problem is with conversion from esmodules to commonjs, not a bug. 

You either need to use `import * as yargs from "yargs"` or need the `allowSyntheticDefaultImports` for typescript.

Read more here:
https://www.typescriptlang.org/tsconfig#allowSyntheticDefaultImports

Tried to explain it for this specific scenario since it's a bit unique but the first link should give you enough info:
TL;DR: You are actually accessing [yargs.default()](https://yargs.js.org/docs/#api-reference-defaultkey-value-description), not `yargs`.

Basically `require("yargs");`, exports the function and all the methods the api uses like `.command()` or `.string()`. 
When you are importing it with typescript by `import yargs from "yargs"`, if the file is commonjs, it calls 'default' on it.

so, this code:

```js
import yargs from "yargs";
yargs(['bar', '-a', '10']).strictOptions().argv;
```
turns into:
```js
const yargs_1 = require("yargs");
yargs_1['default'](['bar', '-a', '10']).strictOptions().argv;
```

Like I mentioned before `require("yargs")` already exports the main function and all the methods, so we don't need to access 'default' on it.
Of course typescript doesn't know it so it just has to assume. 

So it's actually accessing `yargs.default`. This fails in most libraries and will give an error, since they don't export a property called `default`, but yargs actually does.

https://yargs.js.org/docs/#api-reference-defaultkey-value-description

It returns `Argv` which is why it doesn't throw any errors for using `.strictOptions()`. The syntax is obviously wrong for the method, so I assume that messes up the parser.

When you do `import * as yargs from 'yargs'`, it's explicit that you want to import the function all the properties of it, and so it uses the code correctly.

**Comment by loynoir** [comment]
Oh, didn't notice there is a `.default` method.

**Comment by loynoir** [comment]
I think it might be better to rename `.default` method to something else.
As `.default` seems semi-keyword for era among esm and cjs.

**Comment by loynoir** [comment]
Of course, the fastest fix: update document. 🙃

**Comment by karizma** [comment]
I think the readme is correct. Yargs uses different file exports for esm, deno and cjs, so you don't need to use `import * as yargs from "yargs"`.

```js
// file: program.js with { type: "module" } in package.json
import yargs from "yargs";
console.log(yargs.command); // undefined
console.log(yargs().command); // function
```

```js
// file: program.js with cjs
const yargs = require('yargs');
console.log(yargs.command); // function
```

Although I agree,, might wanna change the function name. `defaults()` was a previous name, but another option is `defaultValue()` or `defaultOption()`.

**Comment by bcoe** [comment]
@loynoir, as @karizma says you should be able to use `.defaults`, rather than `default`, and work around your compilation issue.

**Comment by loynoir** [comment]
@karizma @bcoe Anyway, thanks for your reply. Wish you have a nice day.  😃
